perch/model.py

```diff
@@ -303,8 +378,20 @@ def can_create(cls, user, **data):
         raise Return(True)
 
     @coroutine
+    def can_approve(self, user, **data):
+        """Check if a user is authorized to approve a resource"""
+        raise Return(False)
+
+    @coroutine
     def can_update(self, user, **data):
         """Check if a user is authorized to update a resource"""
+
+        # Should only be able to edit resource if in editable state or being changed to editable state
+        current_state_editable = self.state in self.editable_states
+        new_state_editable = 'state' in data and getattr(State, data['state'], None) in self.editable_states
```

Comment:
In fact I think we could even do 
```state =  getattr(State, data.get('state'), self.state)
can_update = state in self.editable_states
````

Since if the user is updating the state we want to make sure that the new state is editable regardless of what the current state is - and if they're not editing the state it will fall back to the current state